### DISCUSSION TITLE
Is anyone aware of a mimic-cxr-ehr-split.csv file in MIMIC-CXR-JPG?

### DISCUSSION THREAD

**Discussion by andrew-w-chen** [MIT-LCP/mimic-code#1840] [MIMIC-IV-CXR] [answered]
I am trying to use and adapt a repository which uses the MIMIC-CXR-JPG database and MIMIC-IV database. The repository can be found here: https://github.com/FirasGit/MeTra/tree/master

In classification/datasets/mimic_lab.py on line 246, the code makes reference to a mimic-cxr-ehr-split.csv file. This file does not appear to be listed in either version 2.0.0 or 2.1.0 of MIMIC-CXR-JPG on physionet.

Interestingly, by googling this file name, I came across what a dissertation written in Russian which also appears to have used this dataset and repository. On page 15 of the dissertation, the file structure of MIMIC-CXR-JPG 2.0.0 is shown and it lists the mimic-cxr-ehr-split.csv. You can read the PDF here: https://sp.susu.ru/student/masterthesis/2024_228_melehinav.pdf.

My question is, does anyone know if the MIMIC-CXR-JPG datasets ever had a mimic-cxr-ehr-splits.csv file and how to obtain it? Or perhaps it is a created file and if it is does anyone know how it is created?

Many thanks.

**Comment by marymlucas** [comment, marked as answer]
That file is not part of the original source files but was created by the MedFuse project that the repo you mentioned references. In the second answer on this issue you can find a link to the code file that generates the splits - [https://github.com/FirasGit/MeTra/issues/4](https://github.com/FirasGit/MeTra/issues/4)

You can also go directly to [https://github.com/nyuad-cai/MedFuse/blob/main/ehr_utils/create_split.py](https://github.com/nyuad-cai/MedFuse/blob/main/ehr_utils/create_split.py)

**Reply by andrew-w-chen** [reply]
Thanks very much!

**Reply by andrew-w-chen** [reply]
Mary, thought I would check here again in case you know the answer to my new problem...

It seems that the mimic3models/split_train_val.py file from [mimic3-benchmark](https://github.com/YerevaNN/mimic3-benchmarks/tree/master) is creating an empty val_filelist.csv

I've been using the mimic3-benchmark repo with the MIMIC-IV dataset since this seemed to be the repo to create files needed to train the original [MeTra](https://github.com/FirasGit/MeTra/tree/master) models. The mimic3-benchmark repo did create all of the correct files in the correct locations from the MIMIC-IV dataset, however, the val_filelist.csv is never populated with any entries when running the split_train_val.py code.

The problem seems to stem from an inconsistency in IDs under resources/valset.csv (which is provided in mimic3-benchmark) and /data/in-hospital-mortality/train/listfile.csv, which was also created by code from mimic3-benchmark. Specifically, The IDs under resources/valset.csv:

100,1
1000,0
10003,0
10004,0
10006,0

Do not seem to match the IDs in /data/in-hospital-mortality/train/listfile.csv:

stay,period_length,stay_id,y_true
11269805_episode2_timeseries.csv,0,36738911,0
17543503_episode1_timeseries.csv,0,34151150,0
16225912_episode1_timeseries.csv,0,32470453,1
15437636_episode1_timeseries.csv,0,39956136,0

If you can provide any insight into how I might mitigate this discrepancy, I would be very appreciative. Again, the original paper said the MIMIC-IV dataset was used and the mimic3-benchmark repo is creating all of the correct files, so I'm not sure what to make of this discrepancy...

**Reply by andrew-w-chen** [reply]
Very shortly after I posted my question here, I found the answer and it seems that it again lies in the MedFuse repo... I don't know how I ended up using the mimic3-benchmark repo and missed that the MedFuse repo has files from mimic3-benchmark modified from MIMIC IV. But all seems to be in order now.
